src/connector/src/parser/mod.rs

```diff
@@ -312,37 +316,70 @@ impl SourceStreamChunkRowWriter<'_> {
         mut f: impl FnMut(&SourceColumnDesc) -> AccessResult<A::Output>,
     ) -> AccessResult<()> {
         let mut wrapped_f = |desc: &SourceColumnDesc| {
-            if let Some(meta_value) =
-                (self.row_meta.as_ref()).and_then(|row_meta| row_meta.value_for_column(desc))
-            {
-                // For meta columns, fill in the meta data.
-                Ok(A::output_for(meta_value))
-            } else {
-                // For normal columns, call the user provided closure.
-                match f(desc) {
-                    Ok(output) => Ok(output),
-
-                    // Throw error for failed access to primary key columns.
-                    Err(e) if desc.is_pk => Err(e),
-                    // Ignore error for other columns and fill in `NULL` instead.
-                    Err(error) => {
-                        // TODO: figure out a way to fill in not-null default value if user specifies one
-                        // TODO: decide whether the error should not be ignored (e.g., even not a valid Debezium message)
-                        // TODO: not using tracing span to provide `split_id` and `offset` due to performance concern,
-                        //       see #13105
-                        static LOG_SUPPERSSER: LazyLock<LogSuppresser> =
-                            LazyLock::new(LogSuppresser::default);
-                        if let Ok(suppressed_count) = LOG_SUPPERSSER.check() {
-                            tracing::warn!(
-                                %error,
-                                split_id = self.row_meta.as_ref().map(|m| m.split_id),
-                                offset = self.row_meta.as_ref().map(|m| m.offset),
-                                column = desc.name,
-                                suppressed_count,
-                                "failed to parse non-pk column, padding with `NULL`"
-                            );
+            match (&desc.column_type, &desc.additional_column_type) {
+                (&SourceColumnType::Offset | &SourceColumnType::RowId, _) => {
+                    // SourceColumnType is for CDC source only.
```

Comment:
This makes me wondering whether we are going to support `include` for CDC sources... Seems not discussed in RFC.

Reply:
Oh, I noticed in slack that we decided to delay it until needed.

 https://risingwave-labs.slack.com/archives/C067SM0GMUN/p1701339228481349?thread_ts=1701338126.236529&cid=C067SM0GMUN

Reply:
It is two separate things actually. 
The code here is for direct-cdc tables, which uses `column_type` to mark the metadata column.
For mq related sources, we use `additional_column_type`.

Reply:
Yes. I know how it works currently. It just makes me pondering that stuff.

Reply:
> It just makes me pondering that stuff.

yes it can be merged but I chose to add a new one to prevent mixing logic with CDC stuff.